### ISSUE TITLE
Prometheus Node Exporter doesn't output interface stats

### ISSUE THREAD

**Issue by clerie** [NixOS/nixpkgs#199201]
### Describe the bug

Prometheus Node Exporter doesn't provide `node_network_receive_bytes_total` and some other anymore.

The logs (`journalctl -eu prometheus-node-exporter`) show the following:

```
Nov 02 18:12:34 gatekeeper node_exporter[830]: ts=2022-11-02T17:12:34.848Z caller=collector.go:169 level=error msg="collector failed" name=netdev duration_seconds=1.7679e-05 err="couldn't get netstats: socket: address family not supported by protocol"
```

### Steps To Reproduce

Run Prometheus Node Exporter:
```
services.prometheus.exporters.node.enable = true;
```

Show metric:

```
curl http://localhost:9100/metrics | grep node_network_receive_bytes_total
```

It will not output anything, but something like this should appear:

```
# TYPE node_network_receive_bytes_total counter
node_network_receive_bytes_total{device="ens3"} 1.43789436e+08
node_network_receive_bytes_total{device="lo"} 181738
node_network_receive_bytes_total{device="wg-monitoring"} 4.682062e+07
node_network_receive_bytes_total{device="wg-vpn"} 0
```

### Fix

The systemd unit of Prometheus Node Exporter is denying access to `AF_NETLINK`.
I can can override the systemd unit and it works again.

```
systemd.services."prometheus-node-exporter".serviceConfig.RestrictAddressFamilies = [ "AF_NETLINK" ];
```

### Notify maintainers

@WilliButz 

### Metadata
Using nixpkgs 412b9917cea092f3d39f9cd5dead4effd5bc4053

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`                            
 - host os: `Linux 5.15.75, NixOS, 22.11 (Raccoon)`                                                                    
 - multi-user?: `yes`                                                                                                  
 - sandbox: `yes`                                      
 - version: `nix-env (Nix) 2.11.0`                    
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by zopieux** [comment]
Yup, confirmed, updated to nixos-unstable and ran into this error. I can confirm the fix works, thanks for providing it.

FYI this seems to be a recurring theme in NixOS, people being [overzealous with systemd hardening](https://github.com/NixOS/nixpkgs/pull/141823/files) and throwing all the possible restrictions without thinking much about the consequences for users. Please be mindful when tweaking systemd service restrictions.

---

![annotely_image](https://user-images.githubusercontent.com/81353/200137496-c05d25a6-a62f-4040-81f0-c95031eb83ed.png)

**Comment by fpletz** [comment]
Fixed by #198638.
